### ISSUE TITLE
About the dropout

### ISSUE THREAD

**Issue by yyaodong** [fchollet/keras#217]
Hi fchollet,

I want to ask whether the dropout layer perform the dropout on batch-wise or sample-wise? In the keras.io, it says dropout is conducted in each update, so presumably samples in one minibatch share the same dropout mask rather than each sample has its own dropout mask?

Cheers

**Comment by phreeza** [comment]
The dropout mask is constructed on each update, but it is independent across samples. The relevant line is here:

https://github.com/fchollet/keras/blob/master/keras/layers/core.py#L154

**Comment by fchollet** [comment]
Yes, for all intents and purposes you can consider that Dropout is per-sample. The Dropout mask is a tensor that is random over every dimension.
